src/paas_charm/charm.py

```diff
@@ -349,6 +482,12 @@ def _create_charm_state(self) -> CharmState:
                 for k, v in charm_config.items()
             },
         )
+        tracing_relation_data = None
+        if self._tracing and self._tracing.is_ready():
+            tracing_relation_data = TempoParameters(
+                endpoint=f'{self._tracing.get_endpoint(protocol="otlp_http")}',
+                service_name=self.app.name,
+            )
```

Comment:
Instead of this, I would pass the tracing object and the app name to `from_charm` rather than this object